action.yml

```diff
@@ -0,0 +1,169 @@
+    required: false
+    default: false
+  release_description:
+    type: string
+    required: false
+    default: ''
+
+runs:
+  using: composite
+  steps:
+
+    - name: Validate inputs
+      shell: bash
+      env:
+        TAG: ${{ inputs.tag }}
+      run: |
+        git check-ref-format "tags/$TAG" > /dev/null
+        if [[ $? != "0" ]]; then
+          echo "Invalid tag"
+          exit 1
+        fi
+
+    - name: Delete existing release if one exists
+      if: ${{ inputs.release == 'true' && inputs.delete_existing == 'true' }}
+      shell: bash
+      env:
+        TAG: ${{ inputs.tag }}
+        GITHUB_REPOSITORY: ${{ github.repository }}
+      run: |
+        # Get id for an existing release matching $TAG
+        # https://docs.github.com/en/rest/releases/releases#get-a-release-by-tag-name
+        RESP_CODE=$(curl -w %{http_code} -s -o __response.json \
+        -X GET https://api.github.com/repos/$GITHUB_REPOSITORY/releases/tags/$TAG \
+        -H "Accept: application/vnd.github.v3+json")
+        if [[ $RESP_CODE != "200" ]];
+          echo "Unable to check tag status for $TAG"
+          exit 1
+        fi
+        RELEASE_ID=$(jq .id __response.json)
+        if [[ $RELEASE_ID == "null" ]]; then
+          echo "Did not find an existing release for tag $TAG"
+        else
+          # https://docs.github.com/en/rest/releases/releases#delete-a-release
+          RESP_CODE=$(curl -w %{http_code} -s -o /dev/null \
+          -X DELETE https://api.github.com/repos/$GITHUB_REPOSITORY/releases/$RELEASE_ID \
+          -H "Accept: application/vnd.github.v3+json" \
+          -H "Authorization: token ${{ github.token }}"
+          if [[ $RESP_CODE != "204" ]];
+            echo "Unable to delete release $RELEASE_ID for tag $TAG"
+            exit 1
+          fi
+          echo "Deleted existing release $RELEASE_ID for tag $TAG"
+        fi
+
+    - name: Delete existing tag if one exists
+      if: ${{ inputs.delete_existing == 'true' }}
+      shell: bash
+      env:
+        TAG: ${{ inputs.tag }}
+        GITHUB_REPOSITORY: ${{ github.repository }}
+      run: |
+        # Check if tag currently exists
+        # Note: not using https://api.github.com/repos/$GITHUB_REPOSITORY/git/refs/tags/<tag>
+        # because that uses a "starts-with" filter, so it will also match -beta1 and -rc1 tags
+        # https://docs.github.com/en/rest/git/tags
+        RESP_CODE=$(curl -w %{http_code} -s -o __response.json \
+        -X GET https://api.github.com/repos/$GITHUB_REPOSITORY/git/refs/tags \
+        -H "Accept: application/vnd.github.v3+json")
+        if [[ $RESP_CODE != "200" ]];
+          echo "Unable to check tag status"
+          exit 1
+        fi
+        FOUND="true"
+        # Check there are any tags so we can use jq array selector later
+        if [[ $(jq 'map(type)' __response.json) =~ object ]]; then
+          if [[ $(jq '.[].ref == "refs/tags/$TAG"' __response.json) == "true" ]]; then
+            FOUND="false"
```

Comment:
I'm a little confused by this... it reads to me as "If the ref in the response matches our tag, set found to false"... but wouldn't that mean found is specifically true? I.e. we _did_ find a ref that matches our tag?

Reply:
yeah it's non-sensical, I've updated the conditional statement